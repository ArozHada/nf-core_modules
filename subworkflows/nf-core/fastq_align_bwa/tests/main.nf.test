nextflow_workflow {

    name "Test Workflow FASTQ_ALIGN_BWA"
    script "../main.nf"
    workflow "FASTQ_ALIGN_BWA"
    config "./nextflow.config"

    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "fastq_align_bwa"
    tag "subworkflows/fastq_align_bwa"
    tag "bwa/mem"
    tag "subworkflows/bam_sort_stats_samtools"

    setup {
        run("BWA_INDEX") {
            script "../../../../modules/nf-core/bwa/index/main.nf"
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                ]
                """
            }
        }
    }

    test("sarscov2 - fastq - single_end") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true) ]
                ]
                input[1] = BWA_INDEX.out.index
                input[2] = false
                input[3] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

    test("sarscov2 - fastq - paired_end") {

        when {
            workflow {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                    ]
                ]
                input[1] = BWA_INDEX.out.index
                input[2] = false
                input[3] = [[],[]]
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }
}
