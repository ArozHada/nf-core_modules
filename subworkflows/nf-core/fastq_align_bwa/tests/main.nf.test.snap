{
    "sarscov2 - fastq - single_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,3d43027d4163ada97911b814001511e5"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam:md5,168247a829b796004701f03d3cc37c50"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.bai:md5,dfd1e93e5413921a9a5650be7ee115da"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.stats:md5,4560519e539144c826e53c45d12e80e2"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.flagstat:md5,2191911d72575a2358b08b1df64ccb53"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.idxstats:md5,613e048487662c694aa4a2f73ca96a20"
                    ]
                ],
                "7": [
                    "versions.yml:md5,2eb84761e8e4973b6d1063e490c6a593",
                    "versions.yml:md5,86b836ae2c81eb2fe590e2f1dd48a039",
                    "versions.yml:md5,a8d75f95d11a57531798879b36163f78",
                    "versions.yml:md5,b3ab8f398b782e29f6bd6a6ce0ccce7b",
                    "versions.yml:md5,d9d583f452daa18bcff053506f8e382c",
                    "versions.yml:md5,ee28326a907ab3f34ab2241fac2d9396"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.bai:md5,dfd1e93e5413921a9a5650be7ee115da"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam:md5,168247a829b796004701f03d3cc37c50"
                    ]
                ],
                "bam_orig": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.bam:md5,3d43027d4163ada97911b814001511e5"
                    ]
                ],
                "csi": [
                    
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.flagstat:md5,2191911d72575a2358b08b1df64ccb53"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.idxstats:md5,613e048487662c694aa4a2f73ca96a20"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test.sorted.bam.stats:md5,4560519e539144c826e53c45d12e80e2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2eb84761e8e4973b6d1063e490c6a593",
                    "versions.yml:md5,86b836ae2c81eb2fe590e2f1dd48a039",
                    "versions.yml:md5,a8d75f95d11a57531798879b36163f78",
                    "versions.yml:md5,b3ab8f398b782e29f6bd6a6ce0ccce7b",
                    "versions.yml:md5,d9d583f452daa18bcff053506f8e382c",
                    "versions.yml:md5,ee28326a907ab3f34ab2241fac2d9396"
                ]
            }
        ],
        "timestamp": "2023-11-29T15:55:16.659879981"
    },
    "sarscov2 - fastq - paired_end": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,809ccfe4300fa5005a9d0d4dc09b1a36"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,02b9df0fdc0a4f0c2458036f760865b2"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.bai:md5,b007e306f88fc50261ba4eb720e63962"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.stats:md5,b04ac2c84bc54adf07c019f649b12089"
                    ]
                ],
                "5": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.flagstat:md5,18d602435a02a4d721b78d1812622159"
                    ]
                ],
                "6": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.idxstats:md5,85d20a901eef23ca50c323638a2eb602"
                    ]
                ],
                "7": [
                    "versions.yml:md5,2eb84761e8e4973b6d1063e490c6a593",
                    "versions.yml:md5,86b836ae2c81eb2fe590e2f1dd48a039",
                    "versions.yml:md5,a8d75f95d11a57531798879b36163f78",
                    "versions.yml:md5,b3ab8f398b782e29f6bd6a6ce0ccce7b",
                    "versions.yml:md5,d9d583f452daa18bcff053506f8e382c",
                    "versions.yml:md5,ee28326a907ab3f34ab2241fac2d9396"
                ],
                "bai": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.bai:md5,b007e306f88fc50261ba4eb720e63962"
                    ]
                ],
                "bam": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam:md5,02b9df0fdc0a4f0c2458036f760865b2"
                    ]
                ],
                "bam_orig": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.bam:md5,809ccfe4300fa5005a9d0d4dc09b1a36"
                    ]
                ],
                "csi": [
                    
                ],
                "flagstat": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.flagstat:md5,18d602435a02a4d721b78d1812622159"
                    ]
                ],
                "idxstats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.idxstats:md5,85d20a901eef23ca50c323638a2eb602"
                    ]
                ],
                "stats": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.sorted.bam.stats:md5,b04ac2c84bc54adf07c019f649b12089"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2eb84761e8e4973b6d1063e490c6a593",
                    "versions.yml:md5,86b836ae2c81eb2fe590e2f1dd48a039",
                    "versions.yml:md5,a8d75f95d11a57531798879b36163f78",
                    "versions.yml:md5,b3ab8f398b782e29f6bd6a6ce0ccce7b",
                    "versions.yml:md5,d9d583f452daa18bcff053506f8e382c",
                    "versions.yml:md5,ee28326a907ab3f34ab2241fac2d9396"
                ]
            }
        ],
        "timestamp": "2023-11-29T16:00:05.456363416"
    }
}